### ISSUE TITLE
Comment placed at the end of function with trailing comma

### ISSUE THREAD

**Issue by chenglou** [babel/babel#1798]
``` js
f(
  a,
  b,
  // test
);
```

Outputs `// test` somewhere else.

**Comment by sebmck** [comment]
Related #1304.

**Comment by sebmck** [comment]
Wait no. This has to do with comment attachment.

**Comment by hzoo** [comment]
Same as https://github.com/babel-plugins/babel-plugin-flow-comments/issues/4 where comments are output at the end?

**Comment by sebmck** [comment]
@hzoo Nope, unrealted. That's due to the original node that has the comments being destroyed while this is about the comments never being attached at all because of range differences.

**Comment by Dashed** [comment]
related to https://github.com/babel/babel/issues/407 and https://github.com/babel/babel/issues/483

**Comment by hzoo** [comment]
@sebmck Are you looking into adding comment attachment now? If not maybe I could give it a try with some guidance

**Comment by chulkilee** [comment]
This breaks [ng-annottate](https://github.com/olov/ng-annotate) when using `/*@ngInject*/` to annotate a function. Use `ngInject` directive prologue at the beginning of a function instead to avoid it.

**Comment by underbyte** [comment]
looks like #1304 might of fixed this

```
f(
  a,
  b
  // test
);
```

Output:
```
f(a, b
// test
);
```

**Comment by sebmck** [comment]
@underbyte Note the trailing comma in the first example in this issue and the title.

**Comment by underbyte** [comment]
@sebmck sorry forgot about that

**Comment by danez** [comment]
This is fixed in 7.0

https://babeljs.io/repl/build/7.0/#?babili=false&browsers=&build=&builtIns=false&code_lz=GYVwdgxgLglg9mABAZwBTIJSIN4ChGIBOAplCIUngQQOTI0BcidNANPtQPScDqAFgE9EMZIih8RiADYwwxRAjF95cgB5Rps4gH4OAXwDayALoBuXHqA&debug=false&circleciRepo=&evaluate=false&lineWrap=false&presets=&prettier=false&targets=&version=7.0.0-alpha.20
